addons/account_payment/controllers/portal.py

```diff
@@ -8,42 +11,123 @@
 
 class PortalAccount(portal.PortalAccount, PaymentPortal):
 
+    @http.route(['/my/invoices/overdue'], type='http', auth="public", methods=['GET'], website=True)
+    def portal_my_overdue_invoices(self, access_token=None, **kw):
+        overdue_invoices = None
+        if access_token:
+            partner = request.env['account.move'].sudo().search([('access_token', '=', access_token)]).partner_id
+            overdue_invoices = request.env['account.move'].sudo().search(self._get_overdue_invoices_domain() + [('partner_id', '=', partner.id)])
+        else:
+            try:
+                request.env['account.move'].check_access_rights('read')
+                overdue_invoices = request.env['account.move'].search(self._get_overdue_invoices_domain())
+            except (AccessError, MissingError):
+                return request.redirect('/my')
+
+        values = self._overdue_invoices_get_page_view_values(overdue_invoices, access_token, **kw)
+        if access_token and 'post_params' in values:
+            # sudo_payment signals that invoices tokens needn't be validated
+            values['post_params']['sudo_payment'] = True
+        return request.render("account_payment.portal_overdue_invoices_page", values)
+
+    def _overdue_invoices_get_page_view_values(self, overdue_invoices, access_token, **kwargs):
+        values = {'page_name': 'overdue_invoices'}
+
+        if len(overdue_invoices) == 0:
+            return values
+
+        batch = overdue_invoices.mapped('line_ids')
+        total_amount = sum(overdue_invoices.mapped('amount_total'))
+        residual_amount = sum(overdue_invoices.mapped('amount_residual'))
+        payment_date = fields.Date.today()
+        batch_name = request.env['account.payment.register'].get_next_batch_communication(date=payment_date, batch=batch)
+        payment = {
+            'date': payment_date,
+            'reference': batch_name,
+            'amount': total_amount,
+            'currency': overdue_invoices[0].company_currency_id,
+        }
+
+        values |= {'payment': payment}
+
+        # There is an assumption that all invoices have the same partner, company, and currency
+        # It is also assumed that they all have the same access rights
+        first_invoice = overdue_invoices[0]
```

Comment:
variable not needed here IMO

Reply:
it is not needed, but I believe it make the comment written up there easier to understand. If I was not the one writing this code, reading it would be more intuitive with this variable here.